File: app/src/main/java/com/nimtego/plectrum/data/repository/datasource/DiskDataStore.java
package com.nimtego.plectrum.data.repository.datasource;


import com.nimtego.plectrum.data.cache.Cache;
import com.nimtego.plectrum.data.model.rss_itunes.PopularResponse;
import com.nimtego.plectrum.data.model.itunes.AlbumsRepository;
import com.nimtego.plectrum.data.model.itunes.ArtistsRepository;
import com.nimtego.plectrum.data.model.itunes.SongsRepository;
import com.nimtego.plectrum.data.model.wiki.WikiSearchResult;

import io.reactivex.Observable;

public class DiskDataStore<E> implements DataStore {

    private final Cache<E> cache;

    public DiskDataStore(Cache<E> cache) {
        this.cache = cache;
    }

    @Override
    public Observable<WikiSearchResult> wikiSearch(String response) {
        return null;
    }

    @Override
    public Observable<SongsRepository> songs(String request) {
        return null;
    }

    @Override
    public Observable<SongsRepository> songsByIdAlbum(int id) {
        return null;
    }

    @Override
    public Observable<ArtistsRepository> artists(String request) {
        return null;
    }


    @Override
    public Observable<AlbumsRepository> albums(String request) {
        return null;
    }

    @Override
    public Observable<SongsRepository> songById(int id) {
        return null;
    }

    @Override
    public Observable<ArtistsRepository> artistById(int id) {
        return null;
    }


    @Override
    public Observable<AlbumsRepository> album(String response) {
        return null;
    }

    @Override
    public Observable<PopularResponse> recent() {
        return null;
    }

    @Override
    public Observable<PopularResponse> topSong(int size) {
        return null;
    }

    @Override
    public Observable<PopularResponse> topAlbum() {
        return null;
    }

    @Override
    public Observable<PopularResponse> hot() {
        return null;
    }

    @Override
    public Observable<PopularResponse> newMusick() {
        return null;
    }
}

File: app/src/main/java/com/nimtego/plectrum/data/repository/datasource/DiskDataStore.kt
package com.nimtego.plectrum.data.repository.datasource


import com.nimtego.plectrum.data.cache.Cache
import com.nimtego.plectrum.data.model.rss_itunes.PopularResponse
import com.nimtego.plectrum.data.model.itunes.AlbumsRepository
import com.nimtego.plectrum.data.model.itunes.ArtistsRepository
import com.nimtego.plectrum.data.model.itunes.SongsRepository
import com.nimtego.plectrum.data.model.wiki.WikiSearchResult

import io.reactivex.Observable

class DiskDataStore<E>(private val cache: Cache<E>) : DataStore {


    override fun topMovie(): Observable<PopularResponse> {
        //todo
        return Observable.create { PopularResponse() }
    }

    override fun wikiSearch(response: String): Observable<WikiSearchResult> {
        //todo
        return Observable.create { WikiSearchResult() }
    }

    override fun songs(response: String): Observable<SongsRepository> {
        //todo
        return Observable.create { SongsRepository() }
    }

    override fun songsByIdAlbum(id: Int): Observable<SongsRepository> {
        //todo
        return Observable.create { SongsRepository() }
    }

    override fun artists(response: String): Observable<ArtistsRepository> {
        //todo
        return Observable.create { ArtistsRepository() }
    }


    override fun albums(response: String): Observable<AlbumsRepository> {
        //todo
        return Observable.create { AlbumsRepository() }
    }

    override fun songById(id: Int): Observable<SongsRepository> {
        //todo
        return Observable.create { SongsRepository() }
    }

    override fun artistById(id: Int): Observable<ArtistsRepository> {
        //todo
        return Observable.create { ArtistsRepository() }
    }


    override fun album(response: String): Observable<AlbumsRepository> {
        //todo
        return Observable.create { AlbumsRepository() }
    }

    override fun recent(): Observable<PopularResponse> {
        //todo
        return Observable.create { PopularResponse() }
    }

    override fun topSong(size: Int): Observable<PopularResponse> {
        //todo
        return Observable.create { PopularResponse() }
    }

    override fun topAlbum(): Observable<PopularResponse> {
        //todo
        return Observable.create { PopularResponse() }
    }

    override fun hot(): Observable<PopularResponse> {
        //todo
        return Observable.create { PopularResponse() }
    }

    override fun newMusic(): Observable<PopularResponse> {
        //todo
        return Observable.create { PopularResponse() }
    }

    override fun topFreeBooks(): Observable<PopularResponse> {
        //todo
        return Observable.create { PopularResponse() }
    }

    override fun topPaidBooks(): Observable<PopularResponse> {
        //todo
        return Observable.create { PopularResponse() }
    }

}
